inspirehep/modules/multieditor/serializers.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, print_function, division
+
+
+from .actions import Addition, Deletion, Update
+
+match_type_map = {
+    'contains': 'contains',
+    'is equal to': 'exact',
+    'does not exist': 'missing',
+    'matches regular expression': 'regex'
+}
+
+
+def sanitize_user_action(schema, action, conditions=None):
+    """
+    Sanitizes the user action and serializes it in the class format
+    :param action:  use action as received from the frontend
+    :param conditions: conditions of action
+    :return: returns the serialized action
+    """
+    action_kwargs = {}
+    if not conditions:
+        conditions = []
+    keypath = action.get('mainKey', '').split('.')
+    if not check_provided_key_path(keypath, schema):
+        return None
+    if keypath:
+        action_kwargs['keypath'] = keypath
+        action_kwargs['value'] = action.get('value')
+        action_kwargs['match_type'] = match_type_map[action.get('matchType')]
+        action_kwargs['conditions'] = conditions
+        if action.get('updateValue'):
+            action_kwargs['update_value'] = action.get('updateValue')
+    return action_kwargs
```

Comment:
I added a check for the matchtype so if there is a key and a matchtype I consider it a valid action, despite not affecting the records

Reply:
What do you mean? If I use this I get:
```python
> sanitize_user_action(schema={}, action={'mainKey': {'keypath': ''}}) == {}
True
```
So if for whatever reason the frontend submits an actualy invalid action, the sanitize does not find it's bad no?

Reply:
hmmm... nop, `''.split('.')` gives a non-empty list `['']`...

Reply:
It will actually giving you a key error when trying to get the action from the`match_type_map` dict...  Not nice either :/

Reply:
this will give you a strange action though:
```python
sanitize_user_action(
    schema={},
    action={'mainKey': '', 'matchType': 'contains'}},
)
```
That will give:
```
{
    'keypath': [''],
    'value', None,
    'match_type': 'contains',
    'conditions': None,
}
```

Is that a valid one?

Reply:
This will pass throught the key validation of ```is_valid_key_path``` and get rejected as a non valid key thus rejecting the action as well.